src/build_targets.rs

```diff
@@ -69,82 +69,31 @@ impl BuildTargets {
     ) -> anyhow::Result<BuildTargets> {
         let pc = targetdir.join(format!("{}.pc", &capi_config.pkg_config.filename));
         let include = if capi_config.header.enabled && capi_config.header.generation {
-            let mut header_name = PathBuf::from(&capi_config.header.name);
-            header_name.set_extension("h");
-            Some(targetdir.join(&header_name))
+            Some(targetdir.join(&capi_config.header.name).with_extension("h"))
         } else {
             None
         };
 
-        let lib_name = name;
-
-        let os = &target.os;
-        let env = &target.env;
-
-        let (shared_lib, static_lib, impl_lib, debug_info, def) = match (os.as_str(), env.as_str())
-        {
-            ("none", _)
-            | ("linux", _)
-            | ("freebsd", _)
-            | ("dragonfly", _)
-            | ("netbsd", _)
-            | ("android", _)
-            | ("haiku", _)
-            | ("illumos", _)
-            | ("emscripten", _) => {
-                let static_lib = targetdir.join(format!("lib{lib_name}.a"));
-                let shared_lib = targetdir.join(format!("lib{lib_name}.so"));
-                (shared_lib, static_lib, None, None, None)
-            }
-            ("macos", _) | ("ios", _) | ("tvos", _) | ("visionos", _) => {
-                let static_lib = targetdir.join(format!("lib{lib_name}.a"));
-                let shared_lib = targetdir.join(format!("lib{lib_name}.dylib"));
-                (shared_lib, static_lib, None, None, None)
-            }
-            ("windows", env) => {
-                let static_lib = if env == "msvc" {
-                    targetdir.join(format!("{lib_name}.lib"))
-                } else {
-                    targetdir.join(format!("lib{lib_name}.a"))
-                };
-                let shared_lib = targetdir.join(format!("{lib_name}.dll"));
-                let impl_lib = if env == "msvc" {
-                    targetdir.join(format!("{lib_name}.dll.lib"))
-                } else {
-                    targetdir.join(format!("{lib_name}.dll.a"))
-                };
-                let def = targetdir.join(format!("{lib_name}.def"));
-                let pdb = if env == "msvc" {
-                    Some(targetdir.join(format!("{lib_name}.pdb")))
-                } else {
-                    None
-                };
-                (shared_lib, static_lib, Some(impl_lib), pdb, Some(def))
-            }
-            _ => unimplemented!("The target {}-{} is not supported yet", os, env),
-        };
-
-        let static_lib = if libkinds.contains(&"staticlib") {
-            Some(static_lib)
-        } else {
-            None
+        let Some(file_names) = FileNames::from_target(target, name, targetdir) else {
+            unimplemented!(
```

Comment:
This can be an `anyhow!` even if should be caught earlier already.

Reply:
done